src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        new Random();
+        int figureIndex = new Random().nextInt();
+        final ColorSupplier colorSupplier = new ColorSupplier();
+        switch (figureIndex) {
+            case 0:
+                return new Circle(colorSupplier.getRandomColor(), getRandomInt());
+            case 1:
+                return new Square(colorSupplier.getRandomColor(), getRandomInt());
+            case 2:
+                return new RightTriangle(colorSupplier.getRandomColor(),
+                        getRandomInt(), getRandomInt());
+            case 3:
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        getRandomDouble(), getRandomDouble(), getRandomDouble());
+            case 4:
+            default:
+                return new Rectangle(colorSupplier.getRandomColor(),
+                        getRandomDouble(), getRandomDouble());
```

Comment:
create a separate private method for each of figures that returns random figure. For example: 
private Circle getRandomCircle() {
  return new Circle(colorSupplier.getRandomColor(), getRandomInt());
}